src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -211,15 +211,21 @@ protected SentryStackFrame InternalCreateFrame(StackFrame stackFrame, bool deman
         protected virtual MethodBase? GetMethod(StackFrame stackFrame)
             => stackFrame.GetMethod();
 
-        private bool IsSystemModuleName(string? moduleName)
+        /// <summary>
+        /// Inform if the given module name belongs to the System or not.
+        /// </summary>
+        /// <param name="moduleName">The module name to be checked.</param>
+        /// <param name="options">The Sentry options.</param>
+        /// <returns>True if the module belongs to the System, false otherwise.</returns>
+        public static bool IsSystemModuleName(string? moduleName, SentryOptions options)
```

Comment:
It's also an opportunity to rename this. I believe a more appropriate name is: `IsInApp` and take the `SentryStackFrame` instead. Keep the current implementation but reverse the current result.

Taking `SentryStackFrame` as opposed to `string` also to get some flexibilty in the future to decide InApp on other frame information other than "module"

How does that sound @bitsandfoxes @SimonCropp ?

Reply:
`frame.InApp ?? = frame.IsInApp(options); ` might sound a bit odd and `IsInApp` can diverge in value from the InApp set on the frame.

What about `frame.ConfigureAppFrame(_options);` where it sets the InApp value only if not previously set?

I'm checking in this case both `Module` and `Function` since  on the .NET SDK we are setting the `Module` parameter and on Unity we set the `Function` parameter.

Reply:
in my limited understanding of the conversation, i prefer `frame.ConfigureAppFrame(_options)`